### DISCUSSION TITLE
Static File Example Question

### DISCUSSION THREAD

**Discussion by emnul** [tokio-rs/axum#2979] [Q&A] [answered]
### Summary

I'm curious why in `examples/static-file-server` we serve the different Routers produced by `using_*` and the other functions on different ports:

```rust
    tokio::join!(
        serve(using_serve_dir(), 3001),
        serve(using_serve_dir_with_assets_fallback(), 3002),
        serve(using_serve_dir_only_from_root_via_fallback(), 3003),
        serve(using_serve_dir_with_handler_as_service(), 3004),
        serve(two_serve_dirs(), 3005),
        serve(calling_serve_dir_from_a_handler(), 3006),
        serve(using_serve_file_from_a_route(), 3307),
    );
```

Why would someone want to do this rather than use the same port for all of them?

### axum version

0.7.7

**Comment by jplatte** [comment, marked as answer]
I think the idea there is that it's really a couple of seperate but closely related examples that are simply packaged up as one binary.

**Reply by emnul** [reply]
That makes sense! I thought this was likely the case, but wasn't sure. Maybe a quick comment could be added to clarify?
